pkg/network/node/networkpolicy.go

```diff
@@ -399,12 +448,11 @@ func (np *networkPolicyPlugin) selectPodsFromNamespaces(nsLabelSel, podLabelSel
 	}
 
 	namespaces := np.selectNamespacesInternal(nsSel)
-	for _, pod := range np.pods {
-		vnid, exists := namespaces[pod.Namespace]
-		if exists && podSel.Matches(labels.Set(pod.Labels)) {
-			peerFlows = append(peerFlows, fmt.Sprintf("reg0=%d, ip, nw_src=%s, ", vnid, pod.Status.PodIP))
+	for namespace := range namespaces {
+		pods := np.selectPodsInternal(namespace, podSel)
+		for _, podIPs := range pods {
+			peerFlows = append(peerFlows, fmt.Sprintf("reg0=%d, ip, nw_src=%s, ", namespaces[namespace], podIPs))
```

Comment:
oh, I was like "how does that work with multiple pod IPs?" but it's just the variable name that's confusing; it should be `podIP` not `podIPs`

Reply:
done